packages/eslint-plugin/src/rules/block-spacing.ts

```diff
@@ -0,0 +1,166 @@
+      description:
+        'Disallow or enforce spaces inside of blocks after opening block and before closing block',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    fixable: 'whitespace',
+    hasSuggestions: baseRule.meta.hasSuggestions,
+    schema: baseRule.meta.schema,
+    messages: baseRule.meta.messages,
+  },
+  defaultOptions: ['always'],
+
+  create(context, [whenToApplyOption]) {
+    const sourceCode = context.getSourceCode();
+    const baseRules = baseRule.create(context);
+    const always = whenToApplyOption !== 'never';
+    const messageId = always ? 'missing' : 'extra';
+    /**
+     * Gets the open brace token from a given node.
+     * @returns The token of the open brace.
+     */
+    function getOpenBrace(
+      node: TSESTree.TSEnumDeclaration,
+    ): TSESTree.PunctuatorToken {
+      // guaranteed for enums
+      // This is the only change made here from the base rule
+      return sourceCode.getFirstToken(node, {
+        filter: token =>
+          token.type === AST_TOKEN_TYPES.Punctuator && token.value == '{',
+      }) as TSESTree.PunctuatorToken;
+    }
+
+    /**
+     * Checks whether or not:
+     *   - given tokens are on same line.
+     *   - there is/isn't a space between given tokens.
+     * @param left A token to check.
+     * @param right The token which is next to `left`.
+     * @returns
+     *    When the option is `"always"`, `true` if there are one or more spaces between given tokens.
+     *    When the option is `"never"`, `true` if there are not any spaces between given tokens.
+     *    If given tokens are not on same line, it's always `true`.
+     */
+    function isValid(left: TSESTree.Token, right: TSESTree.Token): boolean {
+      return (
+        !util.isTokenOnSameLine(left, right) ||
+        sourceCode.isSpaceBetween!(left, right) === always
+      );
+    }
+
+    /**
+     * Checks and reports invalid spacing style inside braces.
+     * @param {ASTNode} node A BlockStatement/StaticBlock/SwitchStatement node to check.
+     * @returns
+     */
+    function checkSpacingInsideBraces(node: TSESTree.TSEnumDeclaration): void {
+      // Gets braces and the first/last token of content.
+      const openBrace = getOpenBrace(node);
+      const closeBrace = sourceCode.getLastToken(node)!;
+      const firstToken = sourceCode.getTokenAfter(openBrace, {
+        includeComments: true,
+      })!;
+      const lastToken = sourceCode.getTokenBefore(closeBrace, {
+        includeComments: true,
+      })!;
+
+      // Skip if the node is invalid or empty.
+      if (
+        openBrace.type !== AST_TOKEN_TYPES.Punctuator ||
+        openBrace.value !== '{' ||
+        closeBrace.type !== AST_TOKEN_TYPES.Punctuator ||
+        closeBrace.value !== '}' ||
+        firstToken === closeBrace
+      ) {
+        return;
+      }
+
+      // Skip line comments for option never
+      if (!always && firstToken.type === AST_TOKEN_TYPES.Line) {
+        return;
+      }
```

Comment:
Indeed. If this isn't covered by tests, then either:

* We need more tests
* This is unreachable and can be removed.

Reply:
Added more tests, and now this is covered